### ISSUE TITLE
Replace all the print statements with a logger

### ISSUE THREAD

**Issue by dan-garvey** [nod-ai/SHARK#1032]
Instead of printing stuff everywhere we should have a logger that outputs it based on a desired verbosity

I also die every time I see the tensorflow device reg

**Comment by tsabbir96** [comment]
hi @dan-garvey , no need to die everytime you see tensorflow device reg. Could you let me know how I can take care of this issue for you? I would love to contribute here.

**Comment by dan-garvey** [comment]
Thanks for your interest. There is a python library called logging: https://docs.python.org/3/library/logging.html

It would be great if you could find all instances of print in the shark codebase and replace them with an appropriate level 
 of logging call. 
 
 If you want to get fancy you could make a custom shark logger class but any switch is a good one at this point. Happy to answer any questions, and feel free to elect me as a reviewer

I think the tensorflow device init issue has already been addressed iirc. (At least, I haven't died recently)

**Comment by tsabbir96** [comment]
Thanks for the detailed answer @dan-garvey, I will probably work on this later this weekend.

**Comment by tsabbir96** [comment]
@dan-garvey Hi, I tried running the main branch locally, but it is struck at :
```
Obtaining file://.../SHARK
  Installing build dependencies ... |
```

**Comment by monorimet** [comment]
It should take anywhere up to 20min for the setup, does it hang longer than that?

**Comment by tsabbir96** [comment]
Hi @monorimet, yeah sadly, it is still like that. I don't know, probably my device or something else. Here's what the logs from my terminal looks like now.
![Screenshot from 2023-05-19 23-11-29](https://github.com/nod-ai/SHARK/assets/49894149/08e37397-011a-4a51-a4b1-9cd77886b789)

**Comment by tsabbir96** [comment]
@dan-garvey @monorimet, Any idea guys?

**Comment by dan-garvey** [comment]
sorry didnt see this, thanks for the ping. Does your htop/top show the process consuming and cpu resource?

**Comment by tsabbir96** [comment]
Yup finally it worked, I am playing around with the Web UI for now. Just getting the hang of it. Thanks!
